src/components/detailedProduct/detailedProductView.ts

```diff
@@ -55,4 +40,29 @@ export default class DetailedProduct {
 
     return main;
   }
+
+  async getProductInformation() {
+    const result = await this.controller.getProductByID(this.id);
+    if (result) {
+      const productData = {
+        name: result.body.masterData.current.name['en-US'],
+        description: result.body.masterData.current.description?.['en-US'],
+        images: result.body.masterData.current.masterVariant.images?.map((image) => image.url),
+        price: result.body.masterData.current.masterVariant.prices?.[0]?.value.centAmount,
+        discountedPrice: result.body.masterData.current.masterVariant.prices?.[0]?.discounted?.value.centAmount,
+      };
+      console.log(result);
```

Comment:
The getProductInformation method is efficiently fetching and processing product data. Consider removing the console.log(result) statement before production deployment.